src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/CommandItemViewModel.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
+using Microsoft.CmdPal.UI.ViewModels.Models;
+
+namespace Microsoft.CmdPal.UI.ViewModels;
+
+public partial class CommandItemViewModel : ObservableObject
+{
+    private readonly ExtensionObject<ICommandItem> _commandItemModel = new(null);
+
+    // private bool _initialized;
+    public string Name { get; private set; } = string.Empty;
+
+    public string Title { get; private set; } = string.Empty;
+
+    public string Subtitle { get; private set; } = string.Empty;
+
+    public string IconUri { get; private set; } = string.Empty;
+
+    public ExtensionObject<ICommand> Command { get; } = new(null);
+
+    public List<CommandContextItemViewModel> MoreCommands { get; private set; } = [];
+
+    public bool HasMoreCommands => MoreCommands.Count > 0;
+
+    public string SecondaryCommandName => HasMoreCommands ? MoreCommands[0].Name : string.Empty;
+
+    public List<CommandContextItemViewModel> AllCommands
+    {
+        get
+        {
+            var command = _commandItemModel.Unsafe?.Command;
+            var model = new CommandContextItem(command!)
+            {
+            };
+            CommandContextItemViewModel defaultCommand = new(model)
+            {
+                Name = Name,
+                Title = Title,
+                Subtitle = Subtitle,
+                IconUri = IconUri,
+
+                // TODO this probably should just be a CommandContextItemViewModel(CommandItemViewModel) ctor, or a copy ctor or whatever
+            };
+
+            var l = new List<CommandContextItemViewModel> { defaultCommand };
+            l.AddRange(MoreCommands);
+            return l;
+        }
+    }
+
+    public CommandItemViewModel(ExtensionObject<ICommandItem> item)
+    {
+        _commandItemModel = item;
+    }
+
+    public async virtual Task InitializePropertiesAsync()
+    {
+        var t = new Task(() =>
+        {
+            try
+            {
+                Initialize();
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine(ex);
+            }
+        });
+        t.Start();
+        await t;
+    }
+
+    protected virtual void Initialize()
+    {
+        var model = _commandItemModel.Unsafe;
+        if (model == null)
+        {
+            return;
+        }
+
+        Name = model.Command?.Name ?? string.Empty;
+        Title = model.Title;
+        Subtitle = model.Subtitle;
+        IconUri = model.Icon.Icon;
+        MoreCommands = model.MoreCommands
+            .Where(contextItem => contextItem is ICommandContextItem)
+            .Select(contextItem => (contextItem as ICommandContextItem)!)
+            .Select(contextItem => new CommandContextItemViewModel(contextItem))
+            .ToList();
+        MoreCommands.ForEach(contextItem =>
+        {
+            contextItem.InitializePropertiesAsync().ContinueWith(t => { /*TODO this feels dirty*/ });
```

Comment:
I think if we resolve the ownership of ensuring the model is loaded to the View in the page that's first trying to use the parent of these (other comment), then we just call `Initialize` straight for each of these?

Reply:
oh good point, we're already in the async initialize here. That makes things easier